### PR TITLE
KIALI-2503 implement anonymous mode with the auth.strategy setting only

### PR DESCRIPTION
**Pull request by jmazzitelli**
Remove the allow-anonymous flag for the login strategy

https://issues.jboss.org/projects/KIALI/issues/KIALI-2503

### CODE REVIEW

**Comment by rhqci** [comment]
Jenkins CI: kiali-core-pr-e2e-test [#649](https://paste.fedoraproject.org/paste/WihrB3-6tn5hbyzsCoEoaQ)
  * :heavy_check_mark: run-kiali-e2e-tests [#[1375]](https://paste.fedoraproject.org/paste/mUR5lhy9Kjsgq~jgKUEsnA)

**Comment by rhqci** [comment]
Jenkins CI: kiali-core-pr-e2e-test [#651](https://paste.fedoraproject.org/paste/YrZHttRJuevrFlTnMtHj3g)
  * :heavy_check_mark: run-kiali-e2e-tests [#[1381]](https://paste.fedoraproject.org/paste/BYZe1HCu8tEHREY8tnpcWw)

**Comment by rhqci** [comment]
Jenkins CI: kiali-core-pr-e2e-test [#653](https://paste.fedoraproject.org/paste/8B-20lNYI3zoxfNzDttUvw)
  * :heavy_check_mark: run-kiali-e2e-tests [#[1382]](https://paste.fedoraproject.org/paste/vDzWVr8zkEy0AMEcn1jWOQ)

**Comment by rhqci** [comment]
Jenkins CI: kiali-core-pr-e2e-test [#654](https://paste.fedoraproject.org/paste/LR8eglhbeqo7Dl2Ws8QMPg)
  * :heavy_check_mark: run-kiali-e2e-tests [#[1383]](https://paste.fedoraproject.org/paste/zOh2MomBuIdUh38kx9scbA)

File: config/config_test.go [hunk 1]
```diff
@@ -241,17 +241,13 @@ func TestMarshalUnmarshalCredentials(t *testing.T) {
 	if conf.Server.Credentials.Passphrase != "bar" {
 		t.Errorf("Failed to unmarshal password credentials:\n%v", conf)
 	}
-	if conf.Server.Credentials.AllowAnonymous {
-		t.Errorf("Failed to unmarshal credentials - anonymous should be false:\n%v", conf)
-	}
 
 	// see that the AllowAnonymous flag can be true
```
Comment (jshaughn) [inline config/config_test.go]: Bad comment

Comment (jmazzitelli) [reply config/config_test.go]: removed

**Pull request merged**